docs/noise.ipynb

```diff
@@ -62,196 +62,673 @@
+   "metadata": {
+    "id": "473ac025a226"
+   },
+   "source": [
+    "Often circuits are defined with just unitary operations, but we want to simulate them with noise. There are several methods for inserting noise in Cirq."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "d0de2c475f12"
+   },
+   "source": [
+    "For any circuit, the `with_noise` method can be called to insert a channel after every moment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "568530c8707b"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"One method to insert noise in a circuit.\"\"\"\n",
+    "# Define some noiseless circuit.\n",
+    "circuit = cirq.testing.random_circuit(\n",
+    "    qubits=3, n_moments=3, op_density=1, random_state=11\n",
+    ")\n",
+    "\n",
+    "# Display the noiseless circuit.\n",
+    "print(\"Circuit without noise:\")\n",
+    "print(circuit)\n",
+    "\n",
+    "# Add noise to the circuit.\n",
+    "noisy = circuit.with_noise(cirq.depolarize(p=0.01))\n",
+    "\n",
+    "# Display it.\n",
+    "print(\"\\nCircuit with noise:\")\n",
+    "print(noisy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "c2206a7b7dd4"
+   },
+   "source": [
+    "This circuit can then be simulated using the methods described above."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "11b18c640767"
+   },
+   "source": [
+    "Another technique is to pass a noise channel to the density matrix simulator as shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "3d16924842bf"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Define a density matrix simulator with a noise model.\"\"\"\n",
+    "noisy_dsim = cirq.DensityMatrixSimulator(\n",
+    "    noise=cirq.generalized_amplitude_damp(p=0.1, gamma=0.5)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "99ac77a2c896"
+   },
+   "source": [
+    "Other than these general methods, channels can be added to circuits at any moment just as gates are. The channels can be different, be correlated, act on a subset of qubits, be custom defined, etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "de8a285b926c"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Defining a circuit with multiple noisy channels.\"\"\"\n",
+    "qreg = cirq.LineQubit.range(4)\n",
+    "circ = cirq.Circuit(\n",
+    "    cirq.H.on_each(qreg),\n",
+    "    cirq.depolarize(p=0.01).on_each(qreg),\n",
+    "    cirq.qft(*qreg),\n",
+    "    bit_phase_flip.on_each(qreg[1::2]),\n",
+    "    cirq.qft(*qreg, inverse=True),\n",
+    "    cirq.reset(qreg[1]),\n",
+    "    cirq.measure(*qreg),\n",
+    "    cirq.bit_flip(p=0.07).controlled(1).on(*qreg[2:]),\n",
+    ")\n",
+    "\n",
+    "print(\"Circuit with multiple channels:\\n\")\n",
+    "print(circ)"
```

Comment:
one more method could be mentioned here: `Gate.with_probability(p)` creates a probabilistic mixture out of a gate.

Reply:
Nice! Added a note on this but eariler in the tutorial - after the `BitAndPhaseFlip` custom definition as an alternative way to define this.